modules/solid_properties/include/materials/ThermalGraphiteProperties.h

```diff
@@ -0,0 +1,129 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ThermalSolidPropertiesMaterial.h"
+
+/**
+ * Graphite thermal properties as a function of temperature appropriate for
+ * graphite moderator and reflectors \cite baker. This form of graphite is
+ * formed by baking carbon at high temperature, and has different properties
+ * from graphite formed by chemical vapor deposition such as pyrolitic carbon.
+ *
+ * The thermal properties of graphite are strongly dependent on the raw materials,
+ * manufacturing process, and oxidation state, so manufacturer-specific properties,
+ * if available, are preferred to the use of this user object.
+ *
+ * At atmospheric pressure, sublimation begins at approximately 3800 K, while
+ * the lowest pressure at which melting is possible is at 103 atm and 4250 K.
+ * Sublimation at reactor temperatures is possible only when the pressure
+ * is many orders of magnitude lower than atmospheric, and hence these properties
+ * correspond only to solid graphite.
+ */
+class ThermalGraphiteProperties : public ThermalSolidPropertiesMaterial
```

Comment:
Would you be open to renaming the graphite classes to indicate the G348 grade, as you've done for 316 steel? This renaming would create an easy path forward to adding more graphite grades to this module (e.g. the AT101 and G535 we use in some advanced manufacturing simulations)

Reply:
Or maybe I should just revisit these correlations to make sure they all come from the same reference... I don't remember why I originally split them across 3 references. I put together these classes like 4 years ago.

Reply:
You make a good point about the enum option vs. individual classes, and that decision feels like something @bwspenc would ultimately make the call on. I'm not sure how to handle a "generic" graphite vs. specific grades: the more I work with graphite the more I realize the amount of variety, but I suspect my advanced manufacturing simulations are more sensitive than other applications. (These aren't Bison simulations, so they probably don't require the same level of consideration). Generic in the enum (as default) matched with the documentation you've already written is probably the way to go. Thanks for being open to considering more granularity for the grades!

Reply:
I ended up going with the enum route - looks like BISON uses a similar approach for its matrix graphite properties. For now, I just added the H-451 grade, but put the enum in place to make this easy to add other grades.